### ISSUE TITLE
i18n: localizer.format signature must match string.format

### ISSUE THREAD

**Issue by x1B** [LaxarJS/laxar#165]
For historical reasons, the signature of `laxar.i18n.localizer( ... ).format` is different from that of `string.format`. 

The localizer version does not take named arguments and uses varargs for the ordered arguments. Meanwhile, `string.format` takes a one parameter for ordered arguments and one parameter for named arguments.

The localizer method should be changed to match the regular version.

**Comment by x1B** [comment]
## Breaking Change

This patch introduces a _breaking change_ to the `laxar.i18n.localizer( ... ).format` API: 

_Previously,_ this API used _varags_ to receive indexed substitutions and could not process named substitutions:

```javascript
var i18nValue = { 'gsw': 'Merssi, [0]. Allegra! [1]'', 'en': 'Thanks, [0]. Bye! [1]' };
var result = ax.localizer( 'gsw' ).format( i18nValue, 'Joe', 'Pete' );
// result == 'Merssi, Joe. Allegra! Pete';
```

_Now,_ to match `laxar.string.format`, the localizers' format method expects (optional) indexed arguments as an array, and (optional) named arguments as an object:

```javascript
var i18nValue = { 'gsw': 'Merssi, [0]. Allegra! [1]'', 'en': 'Thanks, [0]. Bye! [1]' };
var result = ax.localizer( 'gsw' ).format( i18nValue, [ 'Joe', 'Pete' ] );

// Alternatively, using named substitutions:
var i18nValue = { 'gsw': 'Merssi, [to]. Allegra! [from]'', 'en': 'Thanks, [to]. Bye! [from]' };
var result = ax.localizer( 'gsw' ).format( i18nValue, [], { to: 'Joe', from: 'Pete' } );
// result == 'Merssi, Joe. Allegra! Pete';
```

**Comment by x1B** [comment]
Implemented on `master` (`v1.0.0`).
